### ISSUE TITLE
settings.py TEMPLATES.DIRS not as in tutorial

### ISSUE THREAD

**Issue by novski** [mdn/django-locallibrary-tutorial#53]
I may missted something but i can't find where the tutorial changes the line of settings.py in TEMPLATES.DIRS from 
```
'DIRS': [os.path.join(BASE_DIR, 'templates'),],
```
to
```
'DIRS': ['./templates',],
```
what i find in the repo version of code.

[Heres the link to the part where the tutorial shows the DIRS config.](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Home_page#Configuring_where_to_find_the_templates)
[and here is a link to the line of repo.](https://github.com/mdn/django-locallibrary-tutorial/blob/master/locallibrary/settings.py#L63)
did i miss a ajustment somewhare in the tutorial?

**Comment by hamishwillee** [comment]
There is no such line in the tutorial (i.e. you are following the instructions correctly). 

I think you can either ignore or follow the instruction - for root installation the result is the same, but it might not be the same for some configurations. 

Needs some further reading/experimentation to confirm which approach is better. Leaving open as a bug to close down at some point.

**Comment by novski** [comment]
The Problem is i get a error:
```
TemplateDoesNotExist at /catalog/books/
/locallibrary/templates/catalog/book_list.html (Source does not exist)
```
it seams like the `RedirectView.as_view` does not redirect to `catalog` app path.
Instead it looks in the project folder 
```
locallibrary/templates 
```
not 
```
locallibrary/catalog/templates
```
why?

**Comment by hamishwillee** [comment]
OK, so the way that templates are found is described in django docs here: [Templates > configuration](https://docs.djangoproject.com/en/3.1/topics/templates/#configuration). 
- Django will search in the `[DIRS]` 
-  If you set `'APP_DIRS': True,` then Django will search each application you create for its templates in a subdirectory (e.g. we have the catalog application in our locallibrary app). The location it searches is defined in the backend used for the template engine (jinja2 or django templates). Based on my experimentation it searches in an application sub folder named /templates and /templates/application.
- It uses the first template that has a matching name. 

There is no difference between `'DIRS': [os.path.join(BASE_DIR, 'templates'),],` and `'DIRS': ['./templates',],` - you always run the server from root, so they point to the same location. 

Anyway, in the home doc I have updated it to explain that the `'APP_DIRS': True,` is what is important. If you set DIRS empty everything should still work up to the user authentication. In the user authentication doc it specifies using `'DIRS': [os.path.join(BASE_DIR, 'templates'),],` - so the repo itself has been changed to match that and reduce overall confusion.

Upshot, docs for how templates work are more accurate and have better linking. There is no inconsistency between docs and repo now.
